docs/infra/module-dependencies.md

```diff
@@ -90,3 +90,129 @@ parent -- depends on --> config_value
 data.parent -- depends on --> config_value
 child -- depends on --> data.parent
 ```
+
+### Create "data" modules to reference a group of resources created in another module
+
+Often, a root module B will depend on a group of related resources created in another root module A. Moreover, the resources in A are typically created by calling a reusable module in [infra/modules](/infra/modules/) rather than directly in the root module. In this situation, rather than create a separate config for each dependency, create a data module D that B can call to reference R's resources, and group D and R together in a folder e.g. `/infra/modules/component/resources` and `/infra/modules/component/data`. In order to DRY up how related resources are named, create an static `interface` module that both the `resources` module and `data` module can reference. The `interface` module should typically have only one variable as input, the `name` of the group of resources, and the outputs should be the names of related resources in the group.
+
+```terraform
+# root module A
+
+module "config" {
+  ...
+}
+
+module "component" {
+  source = "../modules/component/resources"
```

Comment:
I would add a clarifying inline comment in here, restating that this block (eg. `modules/component/resources`) is responsible for creating resources, whereas the next similar call (eg. `modules/component/data`) is referencing that same set of resources. This is mentioned elsewhere, but not totally clear and unambiguous when you are at this point in the document.

Reply:
Done. Take a look at the comment and lemme know if that helps